shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java

```diff
@@ -52,17 +96,419 @@ public static void addMetaTag(
         messageHeader.setMeta(meta);
     }
 
-    public static void setMessageTypeCoding(Bundle order, Coding coding) {
-        var messageHeader = findOrInitializeMessageHeader(order);
+    // MSH-3 - Sending Application
+    public static MessageHeader.MessageSourceComponent getMSH3MessageSourceComponent(
+            Bundle bundle) {
+        MessageHeader messageHeader = getMSHMessageHeader(bundle);
+        return messageHeader.getSource();
+    }
+
+    public static void setMSH3MessageSourceComponent(
+            Bundle bundle, MessageHeader.MessageSourceComponent sendingApplication) {
+        MessageHeader messageHeader = getMSHMessageHeader(bundle);
+        messageHeader.setSource(sendingApplication);
+    }
+
+    public static MessageHeader.MessageSourceComponent createMSH3MessageSourceComponent() {
```

Comment:
This is only called by test code.  Let's remove it.  If you want this as a helper method to make testing easier, add it to (or create new) a helper class that is just inside the `test/groovy/` area.

Reply:
Moved to test package